core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1767,8 +1772,15 @@ private[spark] class DAGScheduler(
 
           // TODO: mark the executor as failed only if there were lots of fetch failures on it
           if (bmAddress != null) {
-            val hostToUnregisterOutputs = if (env.blockManager.externalShuffleServiceEnabled &&
-              unRegisterOutputOnHostOnFetchFailure) {
+            val executorDecommissionedWithShuffleLost =
+              taskSchedulerImpl.flatMap(
```

Comment:
This is kind of hacky, why not add a new API to TaskScheduler?

Reply:
I could add a new API to TaskScheduler  :-) Was just trying to avoid any "interface change" by going behind the scenes. But yeah it is hacky  !. will change.